pkg/cmd/issue/edit/edit.go

```diff
@@ -157,59 +173,98 @@ func editRun(opts *EditOptions) error {
 		lookupFields = append(lookupFields, "milestone")
 	}
 
-	issue, repo, err := shared.IssueFromArgWithFields(httpClient, opts.BaseRepo, opts.SelectorArg, lookupFields)
+	// Get all specified issues and make sure they are within the same repo.
+	issues, repo, err := shared.IssuesFromArgsWithFields(httpClient, opts.BaseRepo, opts.SelectorArgs, lookupFields)
 	if err != nil {
 		return err
 	}
 
-	editable.Title.Default = issue.Title
-	editable.Body.Default = issue.Body
-	editable.Assignees.Default = issue.Assignees.Logins()
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	if opts.Workers == 0 {
+		opts.Workers = 10
```

Comment:
As this is an IO bound operation do we need to limit the workers? Seems likely that the user will run into API limits here before they ever reach a Go routine limit.

Reply:
10 workers is used elsewhere throughout the CLI. I figured it was because most services limit the number of concurrent connections. Typically it's 2, and networking stacks will often queue requests but those can timeout. Is that not possible in the CLI?

Reply:
Though, seems in most places a worker count is used more where go routines are pulling off a channel like a queue. Still, to make this work for testing, the worker count has to be 1. If left as the zero value, though, I could skip the call to `SetLimit`.